pkg/tsdb/cloudmonitoring/cloudmonitoring_test.go

```diff
@@ -8,1113 +8,1075 @@ import (
+			err = query.parseResponse(res, data, "")
+			require.NoError(t, err)
+			frames, _ := res.Dataframes.Decoded()
+
+			assert.Equal(t, 3, len(frames))
+			assert.Equal(t, "compute.googleapis.com/instance/cpu/usage_time collector-asia-east-1", frames[0].Fields[1].Name)
+			assert.Equal(t, "compute.googleapis.com/instance/cpu/usage_time collector-europe-west-1", frames[1].Fields[1].Name)
+			assert.Equal(t, "compute.googleapis.com/instance/cpu/usage_time collector-us-east-1", frames[2].Fields[1].Name)
+
+			assert.Equal(t, 3, frames[0].Fields[1].Len())
+			assert.Equal(t, 9.8566497180145, frames[0].Fields[1].At(0))
+			assert.Equal(t, 9.7323568146676, frames[0].Fields[1].At(1))
+			assert.Equal(t, 9.7730520330369, frames[0].Fields[1].At(2))
+
+			labels := res.Meta.Get("labels").Interface().(map[string][]string)
+			require.Nil(t, labels)
+			assert.Equal(t, 3, len(labels["metric.label.instance_name"]))
+			assert.Contains(t, labels["metric.label.instance_name"], "collector-asia-east-1")
+			assert.Contains(t, labels["metric.label.instance_name"], "collector-europe-west-1")
+			assert.Contains(t, labels["metric.label.instance_name"], "collector-us-east-1")
+
+			assert.Equal(t, 3, len(labels["resource.label.zone"]))
+			assert.Contains(t, labels["resource.label.zone"], "asia-east1-a")
+			assert.Contains(t, labels["resource.label.zone"], "europe-west1-b")
+			assert.Contains(t, labels["resource.label.zone"], "us-east1-b")
+
+			assert.Equal(t, 1, len(labels["resource.label.project_id"]))
+			assert.Equal(t, "grafana-prod", labels["resource.label.project_id"][0])
+		})
 
-				res := &tsdb.QueryResult{Meta: simplejson.New(), RefId: "A"}
-				query := &cloudMonitoringTimeSeriesFilter{Params: url.Values{}}
-				err = query.parseResponse(res, data, "")
-				So(err, ShouldBeNil)
-				frames, _ := res.Dataframes.Decoded()
+		t.Run("when data from query with no aggregation and group bys", func(t *testing.T) {
+			data, err := loadTestFile("./test-data/2-series-response-no-agg.json")
+			require.NoError(t, err)
+			assert.Equal(t, 3, len(data.TimeSeries))
+
+			res := &plugins.DataQueryResult{Meta: simplejson.New(), RefID: "A"}
+			query := &cloudMonitoringTimeSeriesFilter{Params: url.Values{}, GroupBys: []string{
+				"metric.label.instance_name", "resource.label.zone",
+			}}
+			err = query.parseResponse(res, data, "")
+			require.NoError(t, err)
+			frames, err := res.Dataframes.Decoded()
+			require.NoError(t, err)
+
+			assert.Equal(t, 3, len(frames))
+			assert.Equal(t, "compute.googleapis.com/instance/cpu/usage_time collector-asia-east-1 asia-east1-a", frames[0].Fields[1].Name)
+			assert.Equal(t, "compute.googleapis.com/instance/cpu/usage_time collector-europe-west-1 europe-west1-b", frames[1].Fields[1].Name)
+			assert.Equal(t, "compute.googleapis.com/instance/cpu/usage_time collector-us-east-1 us-east1-b", frames[2].Fields[1].Name)
+		})
 
-				Convey("Should add labels to metric name", func() {
```

Comment:
I can see the point of reduce the nesting in this file, but shouldn't refactoring the tests be done in a separate PR? This test is broken, and it's pretty hard to determine if it's the TSDB service refactoring or the non related refactoring of the test file that caused the break.

Reply:
@sunker I can try to remove the test changes. I originally made the changes since tests were breaking left and right, and it was actually easier to fix them by first converting them.,,,